include/libsemigroups/fpsemi-examples.hpp

```diff
@@ -22,710 +22,1318 @@
+    //!
+    //! \param l the size of the claimed monoid.
+    //!
+    //! \returns a value of type `Presentation<word_type>`.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except
+    [[nodiscard]] Presentation<word_type>
+    not_renner_type_B_monoid_Gay18(size_t l);
+
+    //! \brief A presentation that incorrectly claims to be the Renner monoid of
+    //! type B.
+    //!
+    //! This functions returns a presentation that incorrectly claims to be the
+    //! Renner monoid of type B, as in Section 3.2 of \cite Godelle2009aa.
+    //!
+    //! \param l the size of the claimed monoid.
+    //!
+    //! \returns a value of type `Presentation<word_type>`.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except
+    [[nodiscard]] Presentation<word_type>
+    not_renner_type_B_monoid_God09(size_t l);
+
+    //! \brief A presentation for the Renner monoid of type D.
+    //!
+    //! This functions returns a presentation for the Renner monoid of type D,
+    //! as in Definition 8.4.22 of \cite Gay2018aa.
+    //!
+    //! \param l the size of the monoid.
+    //!
+    //! \returns a value of type `Presentation<word_type>`.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except
     [[nodiscard]] Presentation<word_type>
-    renner_type_B_monoid(size_t l, int q, author val = author::Any);
-
-    // not_renner_type_b_monoid
-    // when q = 0, Example 7.1.2, of Joel's thesis
-    // when q = 1, is Godelle's presentation from: p40 of
-    // https://www.cambridge.org/core/services/aop-cambridge-core/content/view/B6BAB75BD3463916FEDEC15BEDA724FF/S0004972710000365a.pdf/presentation_for_renner_monoids.pdf
-    // or
-    // https://arxiv.org/abs/0904.0926
+    renner_type_D_monoid_Gay18_a(size_t l);
+
+    //! \brief A presentation for the Renner monoid of type D.
+    //!
+    //! This functions returns a presentation for the Renner monoid of type D,
+    //! as in Theorem 8.4.43 of \cite Gay2018aa.
+    //!
+    //! \param l the size of the monoid.
+    //!
+    //! \returns a value of type `Presentation<word_type>`.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except
     [[nodiscard]] Presentation<word_type>
-    not_renner_type_B_monoid(size_t l, int q, author val = author::Any);
+    renner_type_D_monoid_Gay18_b(size_t l);
 
-    // See Theorem 8.4.43 in Joel's thesis q = 1,
-    // for q = 0, Definition 8.4.22 (author = Gay)
+    //! \brief A presentation that incorrectly claims to be the Renner monoid of
+    //! type D.
+    //!
+    //! This functions returns a presentation that incorrectly claims to be the
+    //! Renner monoid of type D.
+    //!
+    //! \param l the size of the claimed monoid.
+    //!
+    //! \returns a value of type `Presentation<word_type>`.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except
     [[nodiscard]] Presentation<word_type>
-    renner_type_D_monoid(size_t l, int q, author val = author::Any);
+    not_renner_type_D_monoid_Machine(size_t l);
```

Comment:
After a discussion with @james-d-mitchell, we decided we should delete all `Machine` presentations. I have done so for all of them except this one. It had no reference and, from what I can tell, it was an example that we came up with that has actual mathematical significance. If I understand correctly, it is the Iwahori-Hecke deformation 0 analogue of an incorrect presentation given by Godelle for the Renner type D monoid (Godelle's presentation being of Iwahori-Hecke deformation 1).

It's not really clear to me what we should do with this function. It seems sensible to keep, but it requires a better name if we keep it. I'm very open to suggestions :).

Theis

Reply:
Keep q as a parameter and give all references possible